inspirehep/modules/migrator/models.py

```diff
@@ -73,12 +74,20 @@ def from_marcxml(cls, raw_record):
 
         The record must have a ``001`` tag containing the recid, otherwise it raises a ValueError.
         """
+        marc_record = create_record(raw_record, keep_singletons=False)
         try:
-            recid = int(cls.re_recid.search(raw_record).group('recid'))
-        except AttributeError:
+            recid = int(marc_record['001'])
+        except KeyError:
             raise ValueError('The MARCXML record contains no recid or recid is malformed')
         # FIXME also get last_updated from marcxml
         record = cls(recid=recid)
         record.marcxml = raw_record
         record.valid = None
+        record.collection = get_collection(marc_record)
```

Comment:
I would move this to `error`, as you don't need the collection if the record migrates properly.